File: C++/169-majority-element.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int j;
        int count = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (count == 0) j = i;
            count += (nums[i] == nums[j] ? 1 : -1);
        }
        return nums[j];
    }
};

File: JS/169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let res, count = 0;
    for (let num of nums) {
        if (count == 0) res = num;
        if (res == num) count++;
        else count--;
    }
    return res;
};